src/base/os.lua

```diff
@@ -80,6 +80,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH") or ""
+			elseif os.is("cygwin") then
+				formats = { "lib%s.a", "%s.a", "lib%s.dll.a", "%s.dll.a" }
+				path = os.getenv("LD_LIBRARY_PATH") or ""

```

Comment:
I agree with Tom on os.is("posix"), but do we still need a Cygwin identifier too? Those crazy ".dll.a" extensions could be justification.